### PR TITLE
Support https on batch processor

### PR DESCRIPTION
**Pull request by majolo**
Adding https support to batch processor, as per customer request.

I've tested this against a cluster running with a signed TLS cert and it's working. Modified the Istio gateway to not accept http traffic, and the processor fails when use-ssl is false as expected.

### CODE REVIEW

File: python/seldon_core/seldon_client.py [hunk 1]
```diff
@@ -1478,19 +1478,19 @@ def rest_predict_gateway(
             request = prediction_pb2.SeldonMessage(data=datadef, meta=metaKV)
         payload = seldon_message_to_json(request)
 
-    if not headers is None:
+    if headers is not None:
```
Comment (majolo) [inline python/seldon_core/seldon_client.py:1481]: Some drive-by syntax fixes, there are plenty more but didn't want to go down a rabbit hole.

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **majolo**
You can assign the PR to them by writing `/assign @majolo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by cliveseldon** [comment]
Seems good to me. Other have worked on batcher more.

**Comment by seldondev** [comment]
@majolo: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 8c7e335f8b396866e329e91c23be2d670841cc88 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3774-integration-d9fsn) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: python/seldon_core/batch_processor.py [hunk 2]
```diff
@@ -79,6 +81,15 @@ def start_multithreaded_batch_worker(
...
         )
 
+    # Providing call credentials sets the REST transport protocol to https,
+    # so we configure credentials even without a supplied token if use_ssl is set.
+    credentials = None
+    if use_ssl or len(call_credentials_token) > 0:
+        token = None
+        if len(call_credentials_token) > 0:
+            token = call_credentials_token
+        credentials = SeldonCallCredentials(token=token)
```
Comment (axsaucedo) [inline python/seldon_core/batch_processor.py:91]: It seems it may also require the SeldonChannelCredentials, as that would be required for GRPC

**Comment by axsaucedo** [comment]
NIce one - looks good,

/approve

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/3774#issuecomment-986922379 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: doc/source/servers/batch.md [hunk 3]
```diff
@@ -95,14 +95,21 @@ Options:
                                   The log level for the batch processor
   -b, --benchmark                 If true the batch processor will print the
                                   elapsed time taken to run the process
-  -u, --batch-id TEXT             Unique batch ID to identify all datapoints
+  -u, --batch-id TEXT             Unique batch ID to identify all data points
                                   processed in this batch, if not provided is
                                   auto generated
-  -u, --batch-size INTEGER        Batch size greater than 1 can be used to
+  -s, --batch-size INTEGER        Batch size greater than 1 can be used to
```
Comment (RafalSkolasinski) [inline doc/source/servers/batch.md:101]: nice catch, it is indeed `-s` in the code

**Comment by seldondev** [comment]
@majolo: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 817b25e2592c6791c776be3e1a42039a21b93dae | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3774-integration-nmzhg) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
Tested locally with REST and it works well. Still trying to test with GRPC but it may be not working due to unrelated bug. 

There is one issue though, new arguments must be added to the test calls: https://github.com/SeldonIO/seldon-core/blob/bd0d4fe0231f97c572e53ad3bad164fd54af65ac/testing/scripts/test_batch_processor.py#L76-L94

**Comment by majolo** [comment]
Fixed that issue @RafalSkolasinski thanks.

**Comment by majolo** [comment]
Ah @RafalSkolasinski It's auto-merged for some reason
